Dockerfile

```diff
@@ -27,68 +27,7 @@ ARG DB_BACKEND=pebbledb
 ARG CMD_PATH=./beacond/cmd
 
 #######################################################
-###         Stage 1 - Cache Go Modules              ###
-#######################################################
-
-FROM golang:${GO_VERSION}-alpine3.20 AS mod-cache
-
-WORKDIR /workdir
-
-RUN apk add --no-cache git
-
-COPY ./beacond/go.mod ./beacond/go.sum ./beacond/
-COPY ./mod/async/go.mod ./mod/async/
-COPY ./mod/beacon/go.mod ./mod/beacon/go.sum ./mod/beacon/
-COPY ./mod/cli/go.mod ./mod/cli/go.sum ./mod/cli/
-COPY ./mod/consensus/go.mod ./mod/consensus/go.sum ./mod/consensus/
-COPY ./mod/consensus-types/go.mod ./mod/consensus-types/go.sum ./mod/consensus-types/
-COPY ./mod/config/go.mod ./mod/config/go.sum ./mod/config/
-COPY ./mod/da/go.mod ./mod/da/go.sum ./mod/da/
-COPY ./mod/engine-primitives/go.mod ./mod/engine-primitives/go.sum ./mod/engine-primitives/
-COPY ./mod/execution/go.mod ./mod/execution/go.sum ./mod/execution/
-COPY ./mod/log/go.mod ./mod/log/go.sum ./mod/log/
-COPY ./mod/node-api/go.mod ./mod/node-api/go.sum ./mod/node-api/
-COPY ./mod/node-api/engines/go.mod ./mod/node-api/engines/go.sum ./mod/node-api/engines/
-COPY ./mod/node-core/go.mod ./mod/node-core/go.sum ./mod/node-core/
-COPY ./mod/observability/go.mod ./mod/observability/
-COPY ./mod/payload/go.mod ./mod/payload/go.sum ./mod/payload/
-COPY ./mod/primitives/go.mod ./mod/primitives/go.sum ./mod/primitives/
-COPY ./mod/state-transition/go.mod ./mod/state-transition/go.sum ./mod/state-transition/
-COPY ./mod/storage/go.mod ./mod/storage/go.sum ./mod/storage/
-COPY ./mod/errors/go.mod ./mod/errors/go.sum ./mod/errors/
-COPY ./mod/geth-primitives/go.mod ./mod/geth-primitives/go.sum ./mod/geth-primitives/
-COPY ./mod/chain-spec/go.mod ./mod/chain-spec/
-
-RUN go work init && \
-    go work use ./beacond && \
-    go work use ./mod/async && \
-    go work use ./mod/beacon && \
-    go work use ./mod/cli && \
-    go work use ./mod/config && \
-    go work use ./mod/consensus && \
-    go work use ./mod/consensus-types && \
-    go work use ./mod/da && \
-    go work use ./mod/engine-primitives && \
-    go work use ./mod/errors && \
-    go work use ./mod/execution && \
-    go work use ./mod/log && \
-    go work use ./mod/node-api && \
-    go work use ./mod/node-api/engines && \
-    go work use ./mod/node-core && \
-    go work use ./mod/observability && \
-    go work use ./mod/payload && \
-    go work use ./mod/primitives && \
-    go work use ./mod/state-transition && \
-    go work use ./mod/storage && \
-    go work use ./mod/geth-primitives && \
-    go work use ./mod/chain-spec
-
-RUN --mount=type=cache,target=/root/.cache/go-build \
-    --mount=type=cache,target=/root/go/pkg/mod \
-    go mod download
```

Comment:
don't we need to run this anyhow? How is that we can drop it?

Reply:
Good catch, I was wondering myself if this is needed with migrating to single module. When I compared the build job to `main` branch it went from ~50s ([job](https://github.com/berachain/beacon-kit/actions/runs/12125478611/job/33805523378?pr=2199)) to 3s ([job](https://github.com/berachain/beacon-kit/actions/runs/12125478611/job/33805523378?pr=2199)) so expected some caching to be there already and didn't think we needed this (other jobs were also often faster). I'll try enabling it again, it should not get worse though